src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3485,6 +3485,75 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
             return Operation.CreateOperationNone(semanticModel: null, operation.Syntax, operation.ConstantValue, childrenBuilder.ToImmutableAndFree(), operation.IsImplicit);
         }
 
+        public override IOperation VisitInterpolatedString(IInterpolatedStringOperation operation, int? captureIdForResult)
+        {
+            var partsBuilder = ArrayBuilder<IInterpolatedStringContentOperation>.GetInstance(operation.Parts.Length);
+            foreach (IInterpolatedStringContentOperation element in operation.Parts)
+            {
+                IInterpolatedStringContentOperation rewrittenElement;
+                if (element.Kind == OperationKind.Interpolation)
+                {
+                    var interpolation = (IInterpolationOperation)element;
+                    _evalStack.Push(Visit(interpolation.Expression));
+
+                    if (interpolation.Alignment != null)
+                    {
+                        _evalStack.Push(Visit(interpolation.Alignment));
+                    }
+
+                    if (interpolation.FormatString != null)
+                    {
+                        _evalStack.Push(Visit(interpolation.FormatString));
```

Comment:
>interpolation.FormatString [](start = 46, length = 26)

Is ```FormatString``` always a string literal as well? Consider doing the same thing as for ```IInterpolatedStringTextOperation.Text```

Reply:
[BoundStringInsert.Format](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Generated/BoundNodes.xml.Generated.cs,5941) is a BoundExpression and the operation factory also doesn't make this assumption (unlike the `IInterpolatedStringTextOperation.Text` case). Is the suggestion to update the bound node to make this explicit?

---
In reply to: [185937456](https://github.com/dotnet/roslyn/pull/26545#discussion_r185937456) [](ancestors = 185937456)

Reply:
Thanks, seems cleaner to update the C# bound nodes as VB bound node already has a BoundLiteral type for Format.

---
In reply to: [185940479](https://github.com/dotnet/roslyn/pull/26545#discussion_r185940479) [](ancestors = 185940479,185940331,185938422,185937456)

Reply:
It looks like the bound node is a literal on VB side

---
In reply to: [185940479](https://github.com/dotnet/roslyn/pull/26545#discussion_r185940479) [](ancestors = 185940479,185940331,185938422,185937456)

Reply:
In C# bound node it looks like it is always a string literal by construction. I'd say it would be good if you could adjust the bound node shape for C# compiler.

---
In reply to: [185940906](https://github.com/dotnet/roslyn/pull/26545#discussion_r185940906) [](ancestors = 185940906,185940479,185940331,185938422,185937456)